### PR TITLE
Remove python2 support from Mariner

### PR DESCRIPTION
**Pull request by oliviacrain**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Python 2 end-of-life was January 1, 2020. Python 2 no longer receives security updates from upstream, which hurts our security posture. Additionally, many packages are dropping support for Python 2, which makes maintaining Python 2 support a burden on our package upgrade efforts.

So, in light of that, let's remove our `python2` package!

Thanks to a lot of incremental `python2` removal work across the past few months, we are finally in a position where removal is possible. This PR removes the `python2` package, alongside `python2` subpackages and macros.

###### Change Log  

- Remove `python2`
- Remove the python2-only packages `python-setuptools` and `python-pip`
- Remove python2-related RPM/SRPM macros from `mariner-rpm-macros`
- Remove `python2` subpackages from `python-six`, `python-setuptools_scm`
- Remove faulty `python2-setuptools` requirement in `mercurial`

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline build ongoing

### CODE REVIEW

**Comment by oliviacrain** [comment]
Meant to mark as a draft PR- please treat this as such until all the checkboxes are all checked.

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json [hunk 1]
```diff
@@ -3022,7 +3022,6 @@
                 "python-packaging",
                 "python-pam",
                 "python-pbr",
-                "python-pip",
```
Comment (nicogbg) [inline SPECS/LICENSES-AND-NOTICES/data/licenses.json:3025]: don't we still provide python-pip for python3 ?

**Review by nicogbg — APPROVED** [review]

File: SPECS/LICENSES-AND-NOTICES/data/licenses.json (hunk 1 above)
Comment (oliviacrain) [reply SPECS/LICENSES-AND-NOTICES/data/licenses.json:3025]: Yep! It's part of the `python3` spec, so it won't appear in this list or in the `cgmanifest.json` file.

Comment (oliviacrain) [reply SPECS/LICENSES-AND-NOTICES/data/licenses.json:3025]: The same sentiment applies to the `python3-setuptools` subpackage.

**Pull request merged**